CHANGELOG.md

```diff
@@ -14,6 +14,7 @@
 * [CHANGE] Ingester: the `cortex_ingester_tsdb_wal_replay_duration_seconds` metrics has been removed. #4465
 * [CHANGE] Query-frontend: use protobuf internal query result payload format by default. This feature is no longer considered experimental. #4557
 * [CHANGE] Ruler: reject creating federated rule groups while tenant federation is disabled. Previously the rule groups would be silently dropped during bucket sync. #4555
+* [CHANGE] Compactor: the `/api/v1/upload/block/{block}/finish` endpoint now returns a `429` status code when the compactor has reached the limit specified by `-compactor.max-block-upload-validation-concurrency`. #4598
```

Comment:
Nit: it wasn't immediately obvious why this relates to the compactor.  I realise that must be where the feature is implemented, but for the average person reading this something like "Backfill upload" might be better than "Compactor".

Reply:
Good point. I changed it to "TSDB Block Upload" since that's how it's described in our docs